backend/core/src/main/java/org/sonarsource/sonarlint/core/tracking/IssueMatchingService.java

```diff
@@ -256,12 +234,28 @@ private void processEvent(AnalysisFinishedEvent event) {
     client.raiseIssues(new RaiseIssuesParams(configurationScopeId, issuesToRaise, false, event.getAnalysisId()));
   }
 
-  private static Map<URI, List<RaisedIssueDto>> getIssuesToRaise(Map<Path, List<TrackedIssue>> updatedIssues) {
+  @NotNull
+  static Map<Path, List<TrackedIssue>> getNewTrackedIssues(Map<Path, List<RawIssue>> rawIssuesByIdeRelativePath) {
+    Map<Path, List<TrackedIssue>> newIssues;
+    newIssues = rawIssuesByIdeRelativePath.entrySet().stream()
+      .map(e -> Map.entry(e.getKey(), e.getValue().stream()
+        .map(issue -> new TrackedIssue(UUID.randomUUID(), issue.getMessage(), Instant.now(), false,
+          issue.getSeverity(), issue.getRuleType(), issue.getRuleKey(), true,
+          getTextRangeWithHash(issue.getTextRange(), issue.getClientInputFile()),
+          getLineWithHash(issue.getTextRange(), issue.getClientInputFile()), null,
+          issue.getImpacts(), issue.getFlows(), issue.getQuickFixes(), issue.getVulnerabilityProbability(),
+          issue.getRuleDescriptionContextKey(), issue.getCleanCodeAttribute(), issue.getFileUri()))
+        .collect(toList())))
+      .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    return newIssues;
+  }
+
+  public static Map<URI, List<RaisedIssueDto>> getIssuesToRaise(Map<Path, List<TrackedIssue>> updatedIssues) {
     return updatedIssues.values().stream().flatMap(Collection::stream)
       .collect(groupingBy(TrackedIssue::getFileUri, Collectors.mapping(DtoMapper::toRaisedIssueDto, Collectors.toList())));
   }
 
-  public void storeTrackedIssues(XodusKnownIssuesStore knownIssuesStore, String configurationScopeId, Path clientRelativePath, List<TrackedIssue> newKnownIssues) {
+  static void storeTrackedIssues(XodusKnownIssuesStore knownIssuesStore, String configurationScopeId, Path clientRelativePath, List<TrackedIssue> newKnownIssues) {
```

Comment:
The fact that we use the same method both for hotspots and issues means that the last one to call it might overwrite what was previously stored. I will push a fix on this PR